### PR TITLE
Add option to better control the tab closing behaviour

### LINKED ISSUE #5967: tabcloseleft option
I found that some time ago someone proposed a patch about the option ``tabcloseleft`` in the [mailing list](https://www.mail-archive.com/vim_dev@googlegroups.com/msg18386.html) and indeed into todo.txt there is the line 

    Patch for :tabcloseleft, after closing a tab go to left tab. (William Bowers, 2012 Aug 4)

and I don't understand if it was dropped or if it needs more work on it in order to be merged or something else.

I think that is an option valuable to add, let me know if I can help, thank for your time.

### PR DESCRIPTION
**Pull request by LemonBoy**
Add the `switchtab` option, loosely modeled after `switchbuf` to control what happens when a tab is closed. At the moment it is possible to specify the next tab to be focused (the default behaviour), the previous one, or the last used one.

Let's bikeshed a (wee) bit about the option (and sub-option) naming.

Close #5967

### CODE REVIEW

File: runtime/doc/options.txt [hunk 1]
```diff
@@ -8059,6 +8059,16 @@ A jump table for the options with a short description can be found at |Q_op|.
...
 
+						*'switchtab'* *'swt'*
+'switchtab' 'swt'	string	(default "")
+			global
+	This option controls the behavior when switching tabs, for example when
+	closing one (e.g.  |:tabclose|).
+	Possible values (comma-separated list):
+	   left		If included, move to the previous tab page instead of the next
+			one.
+	   uselast	If included, jump to the previously used tab page if possible.
```
Comment (chrisbra) [inline runtime/doc/options.txt]: If empty, jump to the next tab page (to the right).

File: runtime/doc/options.txt [hunk 2]
```diff
@@ -8059,6 +8059,16 @@ A jump table for the options with a short description can be found at |Q_op|.
 	If a window has 'winfixbuf' enabled, 'switchbuf' is currently not
 	applied to the split window.
 
+						*'switchtab'* *'swt'*
+'switchtab' 'swt'	string	(default "")
+			global
+	This option controls the behavior when switching tabs, for example when
```
Comment (chrisbra) [inline runtime/doc/options.txt]: Does this really handle *switching tabs*? Isn't it rather about *closing tabs* instead?

**Comment by chrisbra** [comment]
Thanks, but since this is about the behaviour when closing a tab page, shouldn't the option really be called 'tabclose'?

**Comment by LemonBoy** [comment]
> Thanks, but since this is about the behaviour when closing a tab page, shouldn't the option really be called 'tabclose'?

Yes, but it dictates how Vim switches to another page when closing one. I mainly kept the name for symmetry with `switchbuf` but I can change it to `tabclose` if you prefer.

**Comment by chrisbra** [comment]
Yeah, I'd prefer to have it "closer" to the other 'tab' options.

File: runtime/doc/options.txt [hunk 3]
```diff
@@ -8059,6 +8059,16 @@ A jump table for the options with a short description can be found at |Q_op|.
 	If a window has 'winfixbuf' enabled, 'switchbuf' is currently not
 	applied to the split window.
 
+						*'switchtab'* *'swt'*
+'switchtab' 'swt'	string	(default "")
```
Comment (zeertzjq) [inline runtime/doc/options.txt]: The option name is not updated here.

**Comment by chrisbra** [comment]
thanks!

**Pull request closed without merging**